new profiler has failing tests

@jeremyjordan 
Tests fail on OSX

```
tests/test_profiler.py::test_advanced_profiler FAILED 
```

maybe some larger tolerance on the value?
```
        a_duration = get_duration(p.profiled_actions["a"])
>       np.testing.assert_almost_equal(a_duration, [4], decimal=1)
E       AssertionError: 
E       Arrays are not almost equal to 1 decimals
E       
E       Mismatch: 100%
E       Max absolute difference: 0.150021
E       Max relative difference: 0.03750525
E        x: array(4.2)
E        y: array([4])
tests/test_profiler.py:46: AssertionError
```

--- a/tests/test_profiler.py
+++ b/tests/test_profiler.py
@@ -1,6 +1,18 @@
-from pytorch_lightning.profiler import Profiler, AdvancedProfiler
 import time
+
 import numpy as np
+
+from pytorch_lightning.profiler import Profiler, AdvancedProfiler
+
+
+def _assert_almost_equal(got, expect, tol=0.1):
+    expect = np.array(expect)
+    got = np.array(got)
+    is_any_lower = np.any(np.less(got, expect - tol))
+    is_any_above = np.any(np.greater(got, expect + tol))
+    if is_any_lower or is_any_above:
+        raise AssertionError("Expected: %s\nGot: %s\n which is out of the range +/- %f",
+                             expect, got, tol)
 
 
 def test_simple_profiler():
@@ -19,13 +31,14 @@ def test_simple_profiler():
         time.sleep(1)
 
     # different environments have different precision when it comes to time.sleep()
-    np.testing.assert_almost_equal(p.recorded_durations["a"], [3, 1], decimal=1)
-    np.testing.assert_almost_equal(p.recorded_durations["b"], [2], decimal=1)
-    np.testing.assert_almost_equal(p.recorded_durations["c"], [1], decimal=1)
+    # see: https://github.com/PyTorchLightning/pytorch-lightning/issues/796
+    _assert_almost_equal(p.recorded_durations["a"], [3, 1], tol=0.1)
+    _assert_almost_equal(p.recorded_durations["b"], [2], tol=0.1)
+    _assert_almost_equal(p.recorded_durations["c"], [1], tol=0.1)
 
 
 def test_advanced_profiler():
-    def get_duration(profile):
+    def _get_duration(profile):
         return sum([x.totaltime for x in profile.getstats()])
 
     p = AdvancedProfiler()
@@ -42,9 +55,11 @@ def test_advanced_profiler():
     with p.profile("c"):
         time.sleep(1)
 
-    a_duration = get_duration(p.profiled_actions["a"])
-    np.testing.assert_almost_equal(a_duration, [4], decimal=1)
-    b_duration = get_duration(p.profiled_actions["b"])
-    np.testing.assert_almost_equal(b_duration, [2], decimal=1)
-    c_duration = get_duration(p.profiled_actions["c"])
-    np.testing.assert_almost_equal(c_duration, [1], decimal=1)
+    # different environments have different precision when it comes to time.sleep()
+    # see: https://github.com/PyTorchLightning/pytorch-lightning/issues/796
+    a_duration = _get_duration(p.profiled_actions["a"])
+    _assert_almost_equal(a_duration, [4], tol=0.5)
+    b_duration = _get_duration(p.profiled_actions["b"])
+    _assert_almost_equal(b_duration, [2], tol=0.5)
+    c_duration = _get_duration(p.profiled_actions["c"])
+    _assert_almost_equal(c_duration, [1], tol=0.5)

--- a/tests/test_profiler.py
+++ b/tests/test_profiler.py
@@ -5,16 +5,6 @@ import numpy as np
 from pytorch_lightning.profiler import Profiler, AdvancedProfiler
 
 
-def _assert_almost_equal(got, expect, tol=0.1):
-    expect = np.array(expect)
-    got = np.array(got)
-    is_any_lower = np.any(np.less(got, expect - tol))
-    is_any_above = np.any(np.greater(got, expect + tol))
-    if is_any_lower or is_any_above:
-        raise AssertionError("Expected: %s\nGot: %s\n which is out of the range +/- %f",
-                             expect, got, tol)
-
-
 def test_simple_profiler():
     p = Profiler()
 
@@ -32,9 +22,9 @@ def test_simple_profiler():
 
     # different environments have different precision when it comes to time.sleep()
     # see: https://github.com/PyTorchLightning/pytorch-lightning/issues/796
-    _assert_almost_equal(p.recorded_durations["a"], [3, 1], tol=0.1)
-    _assert_almost_equal(p.recorded_durations["b"], [2], tol=0.1)
-    _assert_almost_equal(p.recorded_durations["c"], [1], tol=0.1)
+    np.testing.assert_allclose(p.recorded_durations["a"], [3, 1], atol=0.1)
+    np.testing.assert_allclose(p.recorded_durations["b"], [2], atol=0.1)
+    np.testing.assert_allclose(p.recorded_durations["c"], [1], atol=0.1)
 
 
 def test_advanced_profiler():
@@ -58,8 +48,8 @@ def test_advanced_profiler():
     # different environments have different precision when it comes to time.sleep()
     # see: https://github.com/PyTorchLightning/pytorch-lightning/issues/796
     a_duration = _get_duration(p.profiled_actions["a"])
-    _assert_almost_equal(a_duration, [4], tol=0.5)
+    np.testing.assert_allclose(a_duration, [4], atol=0.5)
     b_duration = _get_duration(p.profiled_actions["b"])
-    _assert_almost_equal(b_duration, [2], tol=0.5)
+    np.testing.assert_allclose(b_duration, [2], atol=0.5)
     c_duration = _get_duration(p.profiled_actions["c"])
-    _assert_almost_equal(c_duration, [1], tol=0.5)
+    np.testing.assert_allclose(c_duration, [1], atol=0.5)

--- a/tests/test_profiler.py
+++ b/tests/test_profiler.py
@@ -22,9 +22,9 @@ def test_simple_profiler():
 
     # different environments have different precision when it comes to time.sleep()
     # see: https://github.com/PyTorchLightning/pytorch-lightning/issues/796
-    np.testing.assert_allclose(p.recorded_durations["a"], [3, 1], atol=0.1)
-    np.testing.assert_allclose(p.recorded_durations["b"], [2], atol=0.1)
-    np.testing.assert_allclose(p.recorded_durations["c"], [1], atol=0.1)
+    np.testing.assert_allclose(p.recorded_durations["a"], [3, 1], rtol=0.2)
+    np.testing.assert_allclose(p.recorded_durations["b"], [2], rtol=0.2)
+    np.testing.assert_allclose(p.recorded_durations["c"], [1], rtol=0.2)
 
 
 def test_advanced_profiler():
@@ -48,8 +48,8 @@ def test_advanced_profiler():
     # different environments have different precision when it comes to time.sleep()
     # see: https://github.com/PyTorchLightning/pytorch-lightning/issues/796
     a_duration = _get_duration(p.profiled_actions["a"])
-    np.testing.assert_allclose(a_duration, [4], atol=0.5)
+    np.testing.assert_allclose(a_duration, [4], rtol=0.2)
     b_duration = _get_duration(p.profiled_actions["b"])
-    np.testing.assert_allclose(b_duration, [2], atol=0.5)
+    np.testing.assert_allclose(b_duration, [2], rtol=0.2)
     c_duration = _get_duration(p.profiled_actions["c"])
-    np.testing.assert_allclose(c_duration, [1], atol=0.5)
+    np.testing.assert_allclose(c_duration, [1], rtol=0.2)

